src/QueryStrategies/BatchQueryStrategies/batch_qs_base.jl

```diff
@@ -0,0 +1,130 @@
+abstract type BatchPQs <: PoolQs end
+abstract type ExtendingBatchQs <: BatchPQs end
+abstract type MultiObjectiveBatchQs <: BatchPQs end
+
+"""
+    select_batch(batch_query_strategy, query_data, label_map, candidate_indices)
+Applies batch_query_strategy to select the most useful unlabeled observations
+from query_data.
+
+label_map is used to find out which observations in query_data are still unlabeled,
+candidate_indices is used to find out which observations in query_data are valid
+candidates for the query.
+
+Returns list of indices of most useful observations.
+"""
+function select_batch end
+
+function get_query_objects(qs::BatchPQs,
+                           query_data::Array{T, 2} where T <: Real,
+                           pools::Vector{Symbol},
+                           global_indices::Vector{Int},
+                           history::Vector{Vector{Int}})::Vector{Int}
+    pool_map = MLLabelUtils.labelmap(pools)
+    haskey(pool_map, :U) || throw(ArgumentError("No more points that are unlabeled."))
+    all_history_values = (length(history) > 0) && (collect(Iterators.flatten(history)))
+    candidate_indices = [i for i in pool_map[:U] if global_indices[i] ∉ all_history_values]
+    debug(getlogger(@__MODULE__), "[QS] Selecting best batch of $(qs.k) from $(length(candidate_indices)) candidates.")
+    local_query_indices = select_batch(qs, query_data, pool_map, candidate_indices)
+    return global_indices[local_query_indices]
+end
+
+"""
+rep_measure computes representativeness of observations.
+Only KDE is supported so far.
+"""
+function get_rep_measure(name::Symbol)::Function
+    if (name == :KDE)
+        return (model::SVDD.OCClassifier, data::Array{T, 2} where T <: Real, labels::Dict{Symbol, Vector{Int}}, candidate_indices::Vector{Int}) -> begin
+            γ = MLKernels.getvalue(model.kernel_fct.alpha)
+            return multi_kde(data, γ)(data[:, candidate_indices])
+        end
+    else
+        return throw(ArgumentError("Invalid representativeness measure $(name) specified."))
+    end
+end
+
+"""
+div_measure computes diversity of all samples to one specific sample
+Iterative computation: only value for added sample needs to be computed, old_scores saves aggregated result
+"""
+function get_iterative_div_measure(name::Symbol)::Function
+    if (name == :AngleDiversity)
+        return (model::SVDD.OCClassifier, data::Array{T, 2} where T <: Real, candidate_indices::Vector{Int}, j::Int, old_scores::Vector{Float64}) -> begin
```

Comment:
The parameter `j` should be renamed to something more intuitive.

Reply:
Yes, I agree and we should also rename `candidate_indices`. If I understand this part correctly we should rename it to:
```
candidate_indices -> batch
j -> candidate
```